packages/sdk-core/src/modules/breadcrumbs/storage/InMemoryBreadcrumbsStorage.ts

```diff
@@ -0,0 +1,58 @@
+import { TimeHelper } from '../../../common/TimeHelper';
+import { AttributeType } from '../../../model/data/BacktraceData';
+import { Breadcrumb } from '../model/Breadcrumb';
+import { BreadcrumbLogLevel } from '../model/BreadcrumbLogLevel';
+import { BreadcrumbType } from '../model/BreadcrumbType';
+import { BreadcrumbsStorage } from './BreadcrumbsStorage';
+
+export class InMemoryBreadcrumbsStorage implements BreadcrumbsStorage {
+    public get lastBreadcrumbId(): number {
+        return this._lastBreadcrumbId;
+    }
+    /**
+     * Breadcrumb name
+     */
+    public readonly name: string = 'bt-breadcrumbs-0';
+
+    private _lastBreadcrumbId: number = TimeHelper.toTimestampInSec(TimeHelper.now());
+    private _breadcrumbs: Breadcrumb[] = [];
+
+    constructor(private readonly _maximumBreadcrumbs: number = 100) {}
+
+    /**
+     * Returns breadcrumbs in the JSON format
+     * @returns Breadcrumbs JSON
+     */
+    public get(): string {
+        return JSON.stringify(this._breadcrumbs);
+    }
+
+    public add(
+        message: string,
+        level: BreadcrumbLogLevel,
+        type: BreadcrumbType,
+        attributes?: Record<string, AttributeType> | undefined,
+    ): number {
+        this._lastBreadcrumbId++;
+        const id = this._lastBreadcrumbId;
+        const breadcrumb: Breadcrumb = {
+            id,
+            message,
+            timestamp: TimeHelper.toTimestampInSec(TimeHelper.now()),
+            type: BreadcrumbType[type].toLowerCase(),
+            level: BreadcrumbLogLevel[level].toLowerCase(),
+        };
+
+        if (attributes) {
+            breadcrumb.attributes = attributes;
+        }
+
+        this._breadcrumbs.push(breadcrumb);
+
+        if (this._maximumBreadcrumbs < this._breadcrumbs.length) {
+            this._breadcrumbs = this._breadcrumbs.slice(this._breadcrumbs.length - this._maximumBreadcrumbs);
+        }
```

Comment:
I think logging will slow drastically when max breadcrumbs are reached due to constant recreating of arrays.

Maybe instead of appending breadcrumb to the end of array, you could begin to overwrite starting elements. When iterating, you would have to remember the index of the first item, which will move everytime when an element is added after the array is filled up.

I understand that this may not be easy, but I worry about our performance here.

Reply:
Breadcrumbs have to be in a specific order. One thing we added in Unity is removing more breadcrumbs than one when we hit the limit. 

I will prepare performance test to see if it impact us and maybe I will remove more than one breadcrumb ? WDTY?

Reply:
Sure.